internal/bminventory/inventory.go

```diff
@@ -383,11 +396,35 @@ func (b *bareMetalInventory) updateImageInfoPostUpload(ctx context.Context, clus
 		cluster.ImageInfo.DownloadURL = signedURL
 	}
 
+	txSuccess := false
+	tx := b.db.Begin()
+	defer func() {
+		if !txSuccess {
+			tx.Rollback()
+		}
+		if r := recover(); r != nil {
+			tx.Rollback()
+		}
+	}()
+
 	dbReply := b.db.Model(&models.Cluster{}).Where("id = ?", cluster.ID.String()).Updates(updates)
 	if dbReply.Error != nil {
 		return errors.New("Failed to generate image: error updating image record")
 	}
 
+	if cluster.ProxyHash != clusterProxyHash {
+		updates["proxy_hash"] = clusterProxyHash
```

Comment:
this line should have been removed. it appears not to update the proxy_hash value and instead the method that did work was 

```
dbReply := b.db.Model(cluster).Update("ProxyHash", clusterProxyHash)
```
I wasn't able to figure out why using `updates["proxy_hash"] = clusterProxyHash` won't update the cluster record.